content/en/docs/contributing/third-party-code-donation.md

```diff
@@ -0,0 +1,80 @@
+---
+title: "Donating Third Party Code to cert-manager"
+linkTitle: "Third Party Code Donations"
+weight: 120
+type: "docs"
+---
+
+The cert-manager project welcomes external contributions and has benefited greatly from thousands
+of commits from hundreds of different contributors. Most code is usually committed through pull
+requests to a specific repo, whether that be the main cert-manager repository or one of the associated
+repositories such as the website.
+
+Some contributions aren't as well suited to that kind of workflow, however. That would most likely
+be because their functionality doesn't belong in any particular existing cert-manager repo, while still
+relating to the cert-manager project.
+
+This document aims to address the donation of code to the cert-manager project, and to provide a
+framework for sustainable contributions which can be tested and relied upon going forwards by both
+cert-manager maintainers and users.
+
+The requirements in this document are based in part on what's done for CoreDNS, Envoy, Kubernetes
+and containerd.
+
+## Requirements
+
+- Code must be licensed appropriately, including any dependencies   
+  We'd prefer [Apache 2.0](https://tldrlegal.com/license/apache-license-2.0-(apache-2.0)) since that's
+  what cert-manager [uses](https://github.com/jetstack/cert-manager/blob/master/LICENSE), but the
+  license must be [OSI approved](https://opensource.org/licenses).
+- Code must conform to CNCF standards and due diligence requirements   
+  You don't need to go over this with a fine-toothed comb; the intent here is that no code donation
+  should have a negative effect on cert-manager's progress as a CNCF project. See the
+  [CNCF due diligence template](https://github.com/cncf/toc/blob/main/process/dd-review-template.md)
+- Must have a sponsor by an existing maintainer   
+  An existing regular contributor to cert-manager must sponsor the adoption of any third party code
+  donation. This ensures that there's a single point of contact for the party donating the code.
+- Must pass cert-manager conformance tests   
+  This might not apply to all donations, but where conformance tests exist any donated code must
+  pass them. E.g. for [external issuers](https://github.com/jetstack/cert-manager/blob/dffbf391dbb0fc6c1cfea62e561a9c6f54362ab0/test/e2e/suite/conformance/certificates/external/external.go#L41-L62)
+- Must provide a point-of-contact for questions about the project for at least 3 months after acceptance
+  We don't anticipate that we'd need to reach out often after the donation has been accepted,
+  but it's important to have someone we can reach out to if we need to.
```

Comment:
Not to be morbid, but what happens if one is offered and then on the 88th day they're gone?

Will you remove the contribution?

Reply:
Yeah I did consider this in a similarly morbid way, but I just figured if that unfortunate situation happened everyone would just be reasonable and work it out; if the project hadn't had any major issues, it would just be accepted and if there were still significant issues we'd just have to discuss it.

If I tried to spend too much time anticipating possibilities like that, I feel the document would grow a lot and could end up being a bit much. Mostly if something diverged from these guidelines for whatever reason, we'd end up chatting about it and working it out in one of the biweekly meetings.